metaflow/client/core.py

```diff
@@ -878,6 +879,53 @@ def tarball(self) -> tarfile.TarFile:
         """
         return self._tar
 
+    @property
+    def extract(self) -> TemporaryDirectory:
+        """
+        Extracts the code package to a temporary directory.
+
+        This creates a temporary directory containing all user code
+        files from the code package. The temporary directory is
+        automatically deleted when the returned TemporaryDirectory
+        object is garbage collected or when its cleanup() is called.
+
+        To preserve the contents to a permanent location, use
+        os.replace() which performs a zero-copy move on the same
+        filesystem:
+
+        ```python
+        with task.code.extract as tmp_dir:
+            # Move contents to permanent location
+            for item in os.listdir(tmp_dir):
+                src = os.path.join(tmp_dir, item)
+                dst = os.path.join('/path/to/permanent/dir', item)
+                os.makedirs(os.path.dirname(dst), exist_ok=True)
+                os.replace(src, dst)  # Atomic move operation
+        ```
+        Returns
+        -------
+        TemporaryDirectory
+            A temporary directory containing the extracted code files.
+            The directory and its contents are automatically deleted when
+            this object is garbage collected.
+        """
+        EXCLUSIONS = [
```

Comment:
i'll also be calling this code in https://github.com/Netflix/metaflow/pull/2238 (will need to factor out into a shared helper method).

apparently the exclusion logic should be

```
    members = [
        m
        for m in tar.getmembers()
        if not any(
            (x.endswith("/") and m.name.startswith(x)) or (m.name == x)
            for x in exclusions
        )
    ]
```

Reply:
ack - fixed.